core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -235,6 +235,11 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.FETCH_SNAPSHOT => requestHelper.closeConnection(request, util.Collections.emptyMap())
         case ApiKeys.BROKER_REGISTRATION => requestHelper.closeConnection(request, util.Collections.emptyMap())
         case ApiKeys.BROKER_HEARTBEAT => requestHelper.closeConnection(request, util.Collections.emptyMap())
+
+        // We need to fix this to forward the request to the active controller.
```

Comment:
We can move this comment to a jira.

Reply:
https://issues.apache.org/jira/browse/KAFKA-12275